flows/definition/legacy.go

```diff
@@ -401,56 +412,78 @@ func createAction(baseLanguage utils.Language, a legacyAction, fieldMap map[stri
 
 func createCase(baseLanguage utils.Language, exitMap map[string]flows.Exit, r legacyRule, translations *flowTranslations) (routers.Case, error) {
 	category := r.Category[baseLanguage]
-	testType := r.Test.Type
+
+	newType, _ := testTypeMappings[r.Test.Type]
 	var arguments []string
 	var err error
 
 	caseUUID := flows.UUID(uuid.NewV4().String())
 
 	switch r.Test.Type {
-	case "subflow":
-		testType = "has_any_word"
-		test := subflowTest{}
-		err = json.Unmarshal(r.Test.Data, &test)
-		if test.ExitType == "completed" {
-			arguments = []string{"completed"}
-		} else {
-			arguments = []string{"errored"}
-		}
-	case "webhook_status":
-		test := webhookTest{}
-		err = json.Unmarshal(r.Test.Data, &test)
-		if test.Status == "success" {
-			testType = "has_webhook_status"
-			arguments = []string{"success"}
-		} else {
-			return routers.Case{}, fmt.Errorf("No failure test")
-		}
-	case "eq":
+
+	// tests that take no arguments
+	case "date", "email", "not_empty", "number", "phone":
+		arguments = []string{}
+
+	// tests against a single numeric value
+	case "eq", "gt", "gte", "lt", "lte":
 		test := stringTest{}
 		err = json.Unmarshal(r.Test.Data, &test)
 		arguments = []string{test.Test}
+
 	case "between":
 		test := betweenTest{}
-		testType = translateTest(testType)
 		err = json.Unmarshal(r.Test.Data, &test)
 		arguments = []string{test.Min, test.Max}
-	case "contains", "contains_any", "contains_phrase", "contains_only_phrase", "regex":
+
+	// tests against a single localized string
+	case "contains", "contains_any", "contains_phrase", "contains_only_phrase", "regex", "starts":
 		test := localizedStringTest{}
 		err = json.Unmarshal(r.Test.Data, &test)
+		arguments = []string{test.Test[baseLanguage]}
 
-		testType = translateTest(testType)
-
-		// TODO: arguments should be an array
 		addTranslationMap(baseLanguage, translations, test.Test, caseUUID, "arguments")
-		arguments = []string{test.Test[baseLanguage]}
+
+	// tests against a single date value
+	case "date_equal", "date_after", "date_before":
+		test := stringTest{}
+		err = json.Unmarshal(r.Test.Data, &test)
+		arguments = []string{test.Test}
+
+	// tests against a single group value
+	case "in_group":
+		test := groupTest{}
+		err = json.Unmarshal(r.Test.Data, &test)
+		arguments = []string{string(test.Test.UUID)}
+
+	case "subflow":
+		newType = "has_any_word"
+		test := subflowTest{}
+		err = json.Unmarshal(r.Test.Data, &test)
+		arguments = []string{test.ExitType}
```

Comment:
I think possible old and new world statuses are the same here - completed or expired